### DISCUSSION TITLE
Is there a way to get a single instance of machine

### DISCUSSION THREAD

**Discussion by piyusharorawork** [statelyai/xstate#2555] [Q&A] [answered]
In React, we have this hook called useMachine which will give create a new instance every time we call it for each component. What if I want only a single instance something like useSingleMachine. 

For me, the use case will be I want only one machine, to have only one context throughout the entire react app. In this way, that context kind of behave like a global state that  we can access the context of that machine from any component

**Comment by ChrisShank** [comment, marked as answer]
Hi! You will want to use [`useInterpret`](https://xstate.js.org/docs/packages/xstate-react/#useinterpret-machine-options-observer) to interpret a machine once. Then you can pass around that service (whose reference never changes so it won't cause unnecessary rerenders) and use [`useActor`](https://xstate.js.org/docs/packages/xstate-react/#api) to subscribe to state changes to that service.

This [article](https://dev.to/mpocock1/how-to-manage-global-state-with-xstate-and-react-3if5) by @mattpocock gives a deeper dive into this particular pattern!

**Reply by piyusharorawork** [reply]
awesome @ChrisShank that is exactly what i was looking for . thanks
